packages/asset-server-plugin/src/s3-asset-storage-strategy.ts

```diff
@@ -230,14 +226,15 @@ export class S3AssetStorageStrategy implements AssetStorageStrategy {
             });
         }
 
-        return body;
+        const typedArray = await body.transformToByteArray();
+        return Readable.from(typedArray)
```

Comment:
In this part, as far as I understand you are "consuming" the stream into a buffer and then creating a new stream from that buffer. That seems redundant to me? Or am I missing something?